### DISCUSSION TITLE
Zoom auf aktuelle Position

### DISCUSSION THREAD

**Discussion by marko-jer** [e-netze/webgis-community#192] [Q&A] [answered]
Hallo!
Ist es möglich, in einer Karte die im MapBuilder als html-Dokument, erstellt wurde,
![image](https://github.com/user-attachments/assets/93d72ce3-0bb5-4a9a-89b1-4a8cda73da7e)
dass beim Aufruf dieser html-Karte mit einem Smartphone
![image](https://github.com/user-attachments/assets/66b10008-11f4-4c63-b461-9f9ef5136723)
auf die aktuelle Position gezoomt wird.
Ich habe in der Dokumentation dazu nix gefunden.
LG
marko

**Comment by gis-eni** [comment]
Im Template sind diese "Actions Beim Start" nicht mit drinnen. Die müsste man manuell einbauen. Dazu baust du im Template innerhalb von ``webgis.init(function() { ... })`` folgendes ein:

``` javascript

if(webgis.isMobileDevice()) {
    var cancelTracker = new webgis.cancelTracker();
    webgis.showProgress('Aktuelle Position wird abgefragt...', null, cancelTracker);

    webgis.currentPosition.get({
        maxWatch: 1,
        onSuccess: function (pos) {
                webgis.hideProgress('Aktuelle Position wird abgefragt...');

                if (!cancelTracker.isCanceled()) {
                    var lng = pos.coords.longitude, lat = pos.coords.latitude, acc = pos.coords.accuracy / (webgis.calc.R) * 180.0 / Math.PI;
                    map.zoomTo([lng - acc, lat - acc, lng + acc, lat + acc]);
                }
        },
        onError: function (err) {
            webgis.hideProgress('Aktuelle Position wird abgefragt...');
        }
    });
}

```

**Reply by marko-jer** [reply, marked as answer]
danke hat funktioniert!
